### ISSUE TITLE
Relative files issue

### ISSUE THREAD

**Issue by XhmikosR** [JustinBeckwith/linkinator#34]
BTW It seems when a link is relative it wrongfully is detected as broken.

For example:

```
Scanning http://localhost:5180/docs/4.3/examples/navbar-static/
 [200] https://getbootstrap.com/docs/4.3/examples/navbar-static/
 [404] http://localhost:5180/navbar-top.css
 [200] http://localhost:5180/docs/4.3/examples/navbar-fixed/
```

`navbar-top.css` is linked in `http://localhost:5180/docs/4.3/examples/navbar-static/` with `<link href="navbar-top.css" rel="stylesheet">`.

At first I thought it was a Windows specific issue but it seems it fails on Travis CI too: https://travis-ci.org/twbs/bootstrap/jobs/494265231#L1399

**Comment by XhmikosR** [comment]
Hey, @JustinBeckwith thanks for the quick patch! Unfortunately, it seems to still fail. :/

See https://travis-ci.org/twbs/bootstrap/jobs/494434529#L1443

**Comment by JustinBeckwith** [comment]
I'm like 99% sure we got it this time 😆 I ran it against the bootstrap generated docs just to make sure I'm not nutz, and everthing looks to work.  Thanks for being patient!

**Comment by XhmikosR** [comment]
Yup, confirmed, thanks!

**Comment by meleyal** [comment]
It seems like this no longer works, I'm getting false positives for relative links (links starting with `../`).

**Comment by samdutton** [comment]
I get the following when running `npx linkinator https://simpl.info --recurse --format CSV`:

```
"https://simpl.info/video/chrome.mp4",404,BROKEN,"https://simpl.info/video",""
"https://simpl.info/video/chrome.webm",404,BROKEN,"https://simpl.info/video",""
"https://simpl.info/images/poster.jpg",404,BROKEN,"https://simpl.info/video",""
```

Here's the HTML:

```
<video autoplay controls playsinline poster="images/poster.jpg" preload="metadata">
  <source src="video/chrome.webm" type="video/webm" />
  <source src="video/chrome.mp4" type="video/mp4" />
  <p>This browser does not support the video element.</p>
</video>
```

I assume the 404 reported by Linkinator is because the `source src` and `poster` values are checked relative to the site root. 

However, I get similar problems with `npx linkinator https://simpl.info/video`, and if I run `npx linkinator .` locally from the `/video` directory, the `source src` and `poster` values work OK, but I get a 404 for `/images/icons/icon192.png` (which Linkinator lists as `"url": "images/icons/icon192.png"` and likewise for an included CSS file.

In other words, I don't understand how to run Linkinator in a way that can handle relative (or root-relative) links.

**Comment by sashasushko** [comment]
@JustinBeckwith due #71 and message from @meleyal – I can give a practical example. We're using the Linkinator in our Neon docs.

On the page https://neon.tech/docs/reference/api-reference/ there are two relative links:
<img width="1177" alt="Снимок экрана 2022-12-13 в 08 36 05" src="https://user-images.githubusercontent.com/4469056/207227747-ee24dc4e-fa76-419c-bfce-c440e8fa1cd8.png">
<img width="1177" alt="Снимок экрана 2022-12-13 в 08 36 15" src="https://user-images.githubusercontent.com/4469056/207227793-18a128dc-e1cc-4d18-8eb5-c9baec419415.png">

You can check, that them are perfectly working in browser. Also, these links are web-correct.

And when I'm checking this page by the Linkinator:

```bash
npm run check:broken-links -- https://neon.tech/docs/reference/api-reference
```

I'm getting errors:

```json
{
  "links": [
    {
      "url": "https://neon.tech/manage/api-keys",
      "status": 404,
      "state": "BROKEN",
      "parent": "https://neon.tech/docs/reference/api-reference"
    },
    {
      "url": "https://neon.tech/manage/branches/",
      "status": 404,
      "state": "BROKEN",
      "parent": "https://neon.tech/docs/reference/api-reference"
    }
  ],
  "passed": false
}
```

But! If I'm passing to the Linkinator the same link with trailing slash:

```bash
npm run check:broken-links -- https://neon.tech/docs/reference/api-reference/
```

There are no errors:

```json
{
  "links": [],
  "passed": true
}
```

**Comment by JustinBeckwith** [comment]
@sashasushko could I trouble you to open this as a new issue?  Makes it much easier to make sure I get around to a fix!

**Comment by sashasushko** [comment]
@JustinBeckwith done – https://github.com/JustinBeckwith/linkinator/issues/529
